### ISSUE TITLE
Changing Customer email address

### ISSUE THREAD

**Issue by michaelbromley** [vendure-ecommerce/vendure#87]
Currently if a Customer edits their details via the [`updateCustomer` mutation](https://www.vendure.io/docs/graphql-api/shop/mutations/#updatecustomer) of the Shop API, the emailAddress field is just patched and updated in the same way as any other field.

Due to the special importance of the emailAddress as the key to the account, additional process is required before allowing a change.

A good discussion can be found here: https://softwareengineering.stackexchange.com/questions/165552/what-is-the-best-policy-for-allowing-clients-to-change-email

In short, the process should go something like this:

1. Customer enters new email address.
2. (if `requireVerification` is set to true) A verification email is sent to the new address
3. Clicking the link in the email completes the process
4. A notification is sent to the old email address in case of a compromised account.

Question: What happens if, after step 2, the customer logs out and then attempts to log in again? Which emailAddress is the valid one? I'd suggest that the old one remains valid until the new one has been verified.
